thales_remote/file_interface.py

```diff
@@ -26,277 +26,328 @@
     """
- 
-    def __init__(self, address, connectionName = "FileExchange"):
-        self._deviceName = connectionName
+
+    _device_name: str
+    remoteConnection: ThalesRemoteConnection
+    _receiver_is_running: bool
+    _automatic_file_exchange: bool
+    _files_to_skip: list[str]
+    receivedFiles: list[dict[str, Union[str, bytearray]]]
+    pathToSave: str
+    _save_received_files_to_disk: bool
+    _keep_files_in_object: bool
+
+    def __init__(self, address, connectionName="FileExchange"):
+        self._device_name = connectionName
         self.remoteConnection = ThalesRemoteConnection()
-        self.remoteConnection.connectToTerm(address, self._deviceName)
+        self.remoteConnection.connectToTerm(address, self._device_name)
         self._receiver_is_running = False
-        self._automaticFileExchange = False
-        self._filesToSkip = ["lastshot.ism"]
+        self._automatic_file_exchange = False
+        self._files_to_skip = ["lastshot.ism"]
         self.receivedFiles = []
         self.pathToSave = os.getcwd()
-        self._saveReceivedFilesToDisk = False
-        self._keepFilesInObject = True
-        return       
-        
-    def close(self):
-        """ Close the file interface.
-        
+        self._save_received_files_to_disk = False
+        self._keep_files_in_object = True
+        return
+
+    def close(self) -> None:
+        """
+        close the file interface
+
         The automatic file sending is disabled and the socket connection is closed.
         """
         self.disableAutomaticFileExchange()
         self.remoteConnection.disconnectFromTerm()
-        return            
-        
-    def enableAutomaticFileExchange(self,enable = True, fileExtensions = "*.ism*.isc*.isw*.iss"):
-        """ Turn on automatic file exchange.
-        
+        return
+
+    def enableAutomaticFileExchange(
+        self, enable: bool = True, fileExtensions: str = "*.ism*.isc*.isw*.iss"
+    ) -> str:
+        """
+        turn on automatic file exchange
+
         If automatic file exchange is enabled, the files configured with the fileExtensions variable are transferred automatically.
-        
+
         In order for the files to be automatically saved to disk, this must be configured with the :func:`~thales_remote.file_interface.ThalesFileInterface.enableSaveReceivedFilesToDisk` method.
         With the method :func:`~thales_remote.file_interface.ThalesFileInterface.enableKeepReceivedFilesInObject`
         can be set whether the files should still remain in the Python object, so that you can read
         the received files as an array with :func:`~thales_remote.file_interface.ThalesFileInterface.getReceivedFiles`.
-        The standard setting is that the files remain in the object.        
-        
+        The standard setting is that the files remain in the object.
+
+        TODO:
+            - The `fileExtensions` parameter looks flimsy; consider using sth. more robust
```

Comment:
Yes this is unfortunately again in connection with the old Thales system the file extensions which are to be sent without separators simply in a string.